### ISSUE TITLE
Mono - System.DllNotFoundException: avutil-55

### ISSUE THREAD

**Issue by J-kit** [filoe/cscore#311]
I was trying to run an application using this lib on debian with mono and it returned this exception:

Unhandled Exception:
System.TypeInitializationException: The type initializer for 'CSCore.Ffmpeg.FfmpegCalls' threw an exception. ---> System.TypeInitializationException: The type initializer for 'CSCore.Ffmpeg.FfmpegUtils' threw an exception. ---> System.DllNotFoundException: avutil-55
  at (wrapper managed-to-native) CSCore.Ffmpeg.Interops.ffmpeg:av_log_set_callback (intptr)
  at CSCore.Ffmpeg.FfmpegCalls.SetLogCallback (CSCore.Ffmpeg.FfmpegCalls+LogCallback callback) [0x00006] in <117ca019c81b40cf9cc0852c6d0ddec3>:0
  at CSCore.Ffmpeg.FfmpegUtils..cctor () [0x00025] in <117ca019c81b40cf9cc0852c6d0ddec3>:0
   --- End of inner exception stack trace ---
  at CSCore.Ffmpeg.Interops.InteropHelper.RegisterLibrariesSearchPath (System.String path) [0x00012] in <117ca019c81b40cf9cc0852c6d0ddec3>:0
  at CSCore.Ffmpeg.FfmpegCalls..cctor () [0x0008f] in <117ca019c81b40cf9cc0852c6d0ddec3>:0
   --- End of inner exception stack trace ---
  at CSCore.Ffmpeg.AvFormatContext..ctor (System.String url) [0x00006] in <117ca019c81b40cf9cc0852c6d0ddec3>:0
  at CSCore.Ffmpeg.FfmpegDecoder..ctor (System.String url) [0x00029] in <117ca019c81b40cf9cc0852c6d0ddec3>:0
  at AudioTool.AudioClient.AudioStart (System.String url) [0x00001] in <57d1cee9c20047c6a0c5acfeeccbf254>:0
  at AudioTool.Program.Main (System.String[] args) [0x00038] in <57d1cee9c20047c6a0c5acfeeccbf254>:0

Any idea how to fix it?

**Comment by filoe** [comment]
Have you had a look at https://github.com/filoe/cscore/blob/master/CSCore.Ffmpeg/Readme.md

**Comment by J-kit** [comment]
Would you mind telling me where i can get the appropriate libraries and how to set the environment variable in debian 8 ?

**Comment by J-kit** [comment]
Would you mind telling me where i can get the appropriate libraries and how to set the environment variable in debian 9 ?

 find -name '*avutil*' returned me a whole bunch of libraries but none of those seem to work
./usr/share/doc/libavutil-dev
./usr/share/doc/libavutil55
./usr/include/x86_64-linux-gnu/libavutil
./usr/include/x86_64-linux-gnu/libavutil/avutil.h
./usr/lib/x86_64-linux-gnu/pkgconfig/libavutil.pc
./usr/lib/x86_64-linux-gnu/libavutil.a
./usr/lib/x86_64-linux-gnu/libavutil.so
./usr/lib/x86_64-linux-gnu/libavutil.so.55
./usr/lib/x86_64-linux-gnu/libavutil.so.55.34.101
./var/cache/apt/archives/libavutil54_6%3a11.11-1~deb8u1_amd64.deb
./var/cache/apt/archives/libavutil-dev_7%3a3.2.9-1~deb9u1_amd64.deb
./var/cache/apt/archives/libavutil-dev_6%3a11.11-1~deb8u1_amd64.deb
./var/cache/apt/archives/libavutil54_6%3a11.9-1~deb8u1_amd64.deb
./var/cache/apt/archives/libavutil55_7%3a3.2.9-1~deb9u1_amd64.deb
./var/cache/apt/archives/libavutil54_6%3a11.8-1~deb8u1_amd64.deb
./var/lib/dpkg/info/libavutil54:amd64.list
./var/lib/dpkg/info/libavutil55:amd64.list
./var/lib/dpkg/info/libavutil54:amd64.postrm
./var/lib/dpkg/info/libavutil-dev:amd64.list
./var/lib/dpkg/info/libavutil55:amd64.md5sums
./var/lib/dpkg/info/libavutil55:amd64.triggers
./var/lib/dpkg/info/libavutil55:amd64.shlibs
./var/lib/dpkg/info/libavutil-dev:amd64.md5sums
./root/tsbot/Debug/FFmpeg/bin/windows/x86/avutil-55.dll
./root/tsbot/Debug/FFmpeg/bin/windows/x64/avutil-55.dll

**Comment by J-kit** [comment]
Ok. Solved issue by installing the appropriate linux packages and copying the libraries from '/usr/lib/x86_64-linux-gnu/' into the folder where the application is. 

After that i had to look what mono expects by setting the mono loglevel to debug (LD_LIBRARY_PATH=./ MONO_LOG_LEVEL=debug  mono MyApp.exe)
An error appeared (something like Mono: DllImport error loading library '/root/bot/Debug/libavutil-55.so': '/root/bot/Debug/libavutil-55.so: cannot open shared object file: No such file or directory'.
)

Then i renamed the existing libavutil.so.55 to libavutil-55.so,I repeated that for every missing include  and mono accepted that.
I hope that helps everyone who has a similar problem in future.
